src/dms/core/EdFi.DataManagementService.Core/OAuthManager.cs

```diff
@@ -0,0 +1,61 @@
+﻿// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+using Microsoft.Extensions.Logging;
+using System.Net;
+using System.Text;
+
+namespace EdFi.DataManagementService.Core;
+
+public interface IOAuthManager
+{
+    public Task<HttpResponseMessage> GetAccessTokenAsync(HttpClient httpClient, string authHeaderString, string upstreamUri);
+}
+public class OAuthManager(ILogger<OAuthManager> logger) : IOAuthManager
+{
+    private readonly ILogger<OAuthManager> _logger = logger;
+    public async Task<HttpResponseMessage> GetAccessTokenAsync(HttpClient httpClient, string authHeaderString, string upstreamUri)
+    {
+        // Generate request message to send upstream
+        _logger.LogInformation("Generating Request message to send upstream");
+        HttpRequestMessage upstreamRequest = new HttpRequestMessage(HttpMethod.Post, upstreamUri);
+
+        // Verify the request header contains Authorization Basic.
+        _logger.LogInformation("Verifying Authorization Header contains 'Basic' Authentication formatting");
+        if (!authHeaderString.Contains("Basic"))
+        {
+            _logger.LogError("Malformed Authorization Header");
+            throw new OAuthIdentityException($"Malformed Authorization Header", HttpStatusCode.BadRequest);
```

Comment:
Since this was ultimately a client error, perhaps it would make sense to simply return a `HttpResponseMessage` instead of throwing an exception? And we can construct it following the `problem+json` approach for error messages.

Add ` TraceId traceId` to the function arguments. Then when we detect the bad content:

```csharp
            return new HttpResponseMessage(HttpStatusCode.BadRequest)
            {
                Content = new StringContent(
                    FailureResponse.ForBadRequest("Malformed Authorization header", traceId, [], []).ToString(),
                    Encoding.UTF8,
                    "application/problem+json"
                ),
            };
```

Do not need to log the error here, since this is a client error, not a system error.